File: CA1/calculator.py
"""
This class contains definitions of the various functions to be included in our calculator programme
"""
#import relevant in-built math functions
import math

#create the Calculator class
class Calculator(object):
    
    #create the init function which is empty to begin with
    def __init__(self):
        pass

    #add two numbers together
    def add(self,first,second): 
        return first + second
        
    #subtract two numbers
    def subtract(self,first,second):
        return first - second

    #multply two numbers
    def multiply(self,first,second):
        return first * second

    #divide two numbers
    def divide(self,first,second):
        if second == 0:
            return 'undefined'
        else:
            return first / float(second)

    #find particular exponential of a number
    def expo(self,number,exponent):
        if exponent == 0:
            return 1
        else:
            return number ** exponent

    #find the square root of a number 
    def square_root(self,number):
        if number <= 0:
            return 'undefined'
        else:
            return number ** (0.5) 
            #return math.sqrt(number)

    #find the square of a number - also handles negative numbers
    def square(self,number):
        return number * number

    #find the cube of a number - also handles negative numbers
    def cube(self,number):
        return (number * number) * number

    #find the sine of a number
    def sine(self,number):
        return round(math.sin(number))
                
    #find the cosine of a number
    def cosine(self,number):
        return round(math.cos(number))

    #find the tangent of a number
    def tangent(self,number):
        return round(math.tan(number))

    #find the sine of a right-angled triangle given lengths of 2 relevant sides
    def sine1(self,opposite,hypoteneuse):
        if opposite <= 0 or hypoteneuse <= 0:
            return 'undefined'
        else: 
            return opposite / float(hypoteneuse)
 
    #find the cosine of a right-angled triangle given lengths of 2 relevant sides
    def cosine1(self,adjacent,hypoteneuse):
        if adjacent <= 0 or hypoteneuse <= 0:
            return 'undefined'
        else:
            return adjacent / float(hypoteneuse)

    #find the tangent of a right-angled triangle given lengths of 2 relevant sides
    def tangent1(self,opposite,adjacent):
        if opposite <= 0 or adjacent <= 0:
            return 'undefined'
        else:
            return opposite / float(adjacent)
    
    
    
    
    
    
    
    

File: CA5/calculator.r
#add two numbers together
add <- function(first, second) {
  return (first + second)
}
add(3, 4)
add(-45, -6)

#subtract two numbers
subtract <- function(first, second) {
  return (first - second)
}
subtract(5, 6)
subtract(-19, -4)

#multply two numbers
multiply <- function(first, second) {
  return (first * second)
}
multiply(10, 4)
multiply(8.2, 6)
multiply(7, 0)

#divide two numbers
divide <- function(first, second) {
  if (second == 0) {
  return ('undefined')
  } else {
  return (first / second)
  }
}
divide(12, 4)
divide(16, 5)
divide(9, 0)

#find particular exponential of a number
expo <- function(number, exponent) {
  if (exponent == 0) {
    return (1)
  } else {
    return (number ** exponent)
  }
}
expo(3, 2)
expo(8, 0)
expo(-4, 2)

#find the square root of a number 
square_root <- function(number) {
  if (number <= 0) {
    return ('undefined')
  } else {
    return (number ** 0.5)
  }
}
square_root(49)
square_root(-5)
square_root(7.8)

#find the square of a number - also handles negative numbers
square <- function(number) {
  return (number * number)
}
square(4.4)
square(-3.4)
square(0)

#find the cube of a number - also handles negative numbers
cube <- function(number) {
  return ((number * number) * number)
}
cube(4)
cube(-5)
cube(2.5)

#find the sine of a number
sine <- function(number) {
  return (sin(number))
}
sine(90)
sine(45)
sine(1)

#find the cosine of a number
cosine <- function(number) {
  return (cos(number))
}
cosine(90)
cosine(45)
cosine(1)

#find the tangent of a number
tangent <- function(number) {
  return (tan(number))
}
tangent(90)
tangent(45)
tangent(1)
  
  